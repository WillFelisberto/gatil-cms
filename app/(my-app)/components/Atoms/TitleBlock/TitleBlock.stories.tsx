import type { Meta, StoryObj } from '@storybook/react';

import { TitleBlock } from './TitleBlock';

const meta: Meta<typeof TitleBlock> = {
  title: 'Atoms/TitleBlock',
  component: TitleBlock,
  tags: ['autodocs'],
  argTypes: {
    tag: {
      control: 'select',
      options: ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']
    },
    alignment: {
      control: 'inline-radio',
      options: ['left', 'center', 'right']
    },
    showIcon: {
      control: 'boolean'
    },
    text: {
      control: 'text'
    },
    id: {
      control: 'text'
    }
  }
};

export default meta;
type Story = StoryObj<typeof TitleBlock>;

export const Default: Story = {
  args: {
    text: 'Título padrão',
    tag: 'h2',
    alignment: 'left',
    showIcon: false,
    blockType: 'title'
  }
};

export const WithIcon: Story = {
  args: {
    ...Default.args,
    text: 'Com ícone',
    showIcon: true
  }
};

export const Centered: Story = {
  args: {
    ...Default.args,
    text: 'Centralizado',
    alignment: 'center'
  }
};

export const H1Tag: Story = {
  args: {
    ...Default.args,
    text: 'Título H1',
    tag: 'h1'
  }
};
