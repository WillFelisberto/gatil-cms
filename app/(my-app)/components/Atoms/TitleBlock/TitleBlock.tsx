import { JSX } from 'react';
import { FaCat } from 'react-icons/fa';

import { BlockProps } from '@/app/(my-app)/types/block-props';

export const TitleBlock = (props: BlockProps<'title'>) => {
  const { text, tag = 'h2', alignment = 'left', showIcon, id } = props;

  const Tag = tag as keyof JSX.IntrinsicElements;

  const alignmentClass = {
    left: 'text-left justify-start',
    center: 'text-center justify-center',
    right: 'text-right justify-end'
  }[alignment];

  return (
    <Tag
      id={id || undefined}
      data-testid="title-block"
      className={`text-3xl md:text-4xl font-bold text-[#013274] mb-12 mt-6 flex items-center gap-2 ${alignmentClass}`}
    >
      {text}
      {showIcon && <FaCat data-testid="title-icon" />}
    </Tag>
  );
};
