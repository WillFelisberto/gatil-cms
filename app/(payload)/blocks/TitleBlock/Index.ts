import { Block } from 'payload';

export const TitleBlock: Block = {
  slug: 'title',
  labels: {
    singular: 'Título',
    plural: 'Títulos'
  },
  fields: [
    {
      name: 'text',
      type: 'text',
      required: true,
      label: 'Texto do título'
    },
    {
      name: 'tag',
      type: 'select',
      label: 'Tag HTML',
      options: ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'],
      defaultValue: 'h2',
      required: true
    },
    {
      name: 'alignment',
      type: 'select',
      label: 'Alinhamento do texto',
      options: [
        { label: 'Esquerda', value: 'left' },
        { label: 'Centro', value: 'center' },
        { label: 'Direita', value: 'right' }
      ],
      defaultValue: 'left',
      required: true
    },
    {
      name: 'showIcon',
      type: 'checkbox',
      label: 'Mostrar icone',
      defaultValue: false
    }
  ]
};
