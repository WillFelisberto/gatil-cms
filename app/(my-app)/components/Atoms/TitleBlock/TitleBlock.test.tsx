import type { BlockProps } from '@/app/(my-app)/types/block-props';
import { render, screen } from '@/tests/test-utils';

import { TitleBlock } from './TitleBlock';

const defaultProps: BlockProps<'title'> = {
  text: 'Título Teste',
  tag: 'h2',
  alignment: 'left',
  blockType: 'title'
};

describe('<TitleBlock />', () => {
  it('Should renders with default h2 tag and text', () => {
    const { container } = render(<TitleBlock {...defaultProps} />);
    const el = screen.getByTestId('title-block');
    expect(el).toBeInTheDocument();
    expect(el.tagName.toLowerCase()).toBe('h2');
    expect(el).toHaveTextContent('Título Teste');

    expect(container).toMatchSnapshot();
  });

  it('Should renders with custom tag (h3)', () => {
    render(<TitleBlock {...defaultProps} tag="h3" />);
    const el = screen.getByTestId('title-block');
    expect(el.tagName.toLowerCase()).toBe('h3');
  });

  it('Should applies correct alignment classes', () => {
    render(<TitleBlock {...defaultProps} alignment="right" />);
    const el = screen.getByTestId('title-block');
    expect(el).toHaveClass('text-right');
    expect(el).toHaveClass('justify-end');
  });

  it('Should renders the icon when showIcon is true', () => {
    render(<TitleBlock {...defaultProps} showIcon />);
    expect(screen.getByTestId('title-icon')).toBeInTheDocument();
  });

  it('Should does not render the icon when showIcon is false', () => {
    render(<TitleBlock {...defaultProps} showIcon={false} />);
    expect(screen.queryByTestId('title-icon')).not.toBeInTheDocument();
  });

  it('Should sets the id attribute if provided', () => {
    render(<TitleBlock {...defaultProps} id="titulo-id" />);
    const el = screen.getByTestId('title-block');
    expect(el).toHaveAttribute('id', 'titulo-id');
  });
});
