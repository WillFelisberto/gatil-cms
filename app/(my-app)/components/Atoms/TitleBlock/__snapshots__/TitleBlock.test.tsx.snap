// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleBlock /> Should renders with default h2 tag and text 1`] = `
<div>
  <h2
    class="text-3xl md:text-4xl font-bold text-[#013274] mb-12 mt-6 flex items-center gap-2 text-left justify-start"
    data-testid="title-block"
  >
    TÃ­tulo Teste
  </h2>
</div>
`;
